core/src/main/java/com/datastax/oss/driver/api/core/CqlSession.java

```diff
@@ -123,6 +123,14 @@ default ResultSet execute(@NonNull String query) {
    *
    * If you want to customize this behavior, you can write your own implementation of {@link
    * PrepareRequest} and pass it to {@link #prepare(PrepareRequest)}.
+   *
+   * <p>The result of this method is cached: if you call it twice with the same {@link
+   * SimpleStatement}, you will get the same {@link PreparedStatement} instance. We still recommend
+   * keeping a reference to it (for example by caching it as a field in a DAO); if that's not
+   * possible (e.g. if query strings are generated dynamically), it's OK to call this method every
+   * time: there will just be a small performance overhead to check the internal cache. Note that
+   * caching is based on the query string exactly as you provided it: the driver does not perform
```

Comment:
We should also indicate that we also cache the statement by more than the query string, but also all statement configuration values.